cli/cli/helpers/portal_manager/portal_manager.go

```diff
@@ -33,15 +46,139 @@ func (portalManager *PortalManager) GetClient() portal_generated_api.KurtosisPor
 }
 
+// CurrentStatus returns the status of any current Kurtosis Portal process running locally
+// It returns:
+// - an int corresponding to the PID of the Portal process as stored in its PID file, 0 otherwise
+// - a pointer to the os.Process object corresponding to the currently running Portal process, nil if none is running
+// - a boolean flag corresponding to whether the Portal process is reachable on its ports
+// - an error if something unexpected happened
+func (portalManager *PortalManager) CurrentStatus(ctx context.Context) (int, *os.Process, bool, error) {
+	pid, err := getPidFromPidFile()
+	if err != nil {
+		return 0, nil, false, stacktrace.Propagate(err, "Unexpected error reading PID from PID file")
+	}
+	if pid == 0 {
+		return 0, nil, false, nil
+	}
+
+	process, err := getRunningProcessFromPid(pid)
+	if err != nil {
+		return 0, nil, false, stacktrace.Propagate(err, "Unexpected error getting process for PID '%d'", pid)
+	}
+	if process == nil {
+		return pid, nil, false, nil
+	}
+
+	if !portalManager.IsReachable() {
+		return pid, process, false, nil
+	}
+	return pid, process, true, nil
+}
+
+// StartNew starts a new Portal process. The caller needs to make sure a process is not already running, as ports
+// might conflict
+// It writes the PID of the new process to the Portal PID file
+// It returns the PID of the new Portal process, or an error if something went wrong
+func (portalManager *PortalManager) StartNew(ctx context.Context) (int, error) {
+	portalPidFilePath, err := host_machine_directories.GetPortalPidFilePath()
+	if err != nil {
+		return 0, stacktrace.Propagate(err, "Unable to get file path to PID file")
+	}
+	portalBinaryFile, err := host_machine_directories.GetPortalBinaryFilePath()
+	if err != nil {
+		return 0, stacktrace.Propagate(err, "Unable to get file path to PID file")
+	}
+
+	// Start portal daemon
+	portalLogFilePath, err := host_machine_directories.GetPortalLogFilePath()
+	if err != nil {
+		return 0, stacktrace.Propagate(err, "Unable to get file path to portal log file")
+	}
+	// Create will truncate the file if it already exists.
+	// TODO: we can potentially do log rolling here
+	portalLogFile, err := os.Create(portalLogFilePath)
+	if err != nil {
+		return 0, stacktrace.Propagate(err, "Error truncating Kurtosis Portal log file prior to starting it")
+	}
+	kurtosisPortalCmd := exec.Command(portalBinaryFile)
+	kurtosisPortalCmd.Stdout = portalLogFile
+	kurtosisPortalCmd.Stderr = portalLogFile
+	if err = kurtosisPortalCmd.Start(); err != nil {
+		// TODO: maybe print the portal logs here
+		return 0, stacktrace.Propagate(err, "Error trying to start Kurtosis Portal")
+	}
+
+	// Persist new PID to PID file
+	newPid := kurtosisPortalCmd.Process.Pid
+	if err = os.WriteFile(portalPidFilePath, []byte(strconv.Itoa(newPid)), portalPidFileMode); err != nil {
+		return 0, stacktrace.Propagate(err, "Portal was successfully started, but Kurtosis failed at persisting its "+
```

Comment:
So, yeah, it's kind of a big deal actually because it means it won't be able to stop it, users will basically need to kill it themselves :/ I'd like to keep the fail loudly here, it shouldn't happen too often anyway, writing to a file shouldn't be too too risky those days. If it becomes too noisy we can revisit. I was contemplating have the PID be served by the daemon itself, as well as its version number. If/When we have this we will be able to write less things on disk, and this will become more robust 👍